docs/aggregation.ipynb

```diff
@@ -1,260 +1,294 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "68d6f586",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import apache_beam as beam\n",
+    "import numpy as np\n",
+    "import xarray_beam as xbeam\n",
+    "import xarray\n",
+    "\n",
+    "ds = xarray.tutorial.load_dataset('air_temperature')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "f18bb1ff",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:apache_beam.coders.coder_impl:Using fallback deterministic coder for type '<class 'xarray_beam._src.core.Key'>' in '[6]: Mean/PerKey/CombinePerKey(MeanCombineFn)/GroupByKey'. \n",
+      "WARNING:apache_beam.coders.coder_impl:Using fallback deterministic coder for type '<class 'xarray_beam._src.core.Key'>' in '[6]: Mean/PerKey/CombinePerKey(MeanCombineFn)/GroupByKey'. \n",
+      "WARNING:apache_beam.coders.coder_impl:Using fallback deterministic coder for type '<class 'xarray_beam._src.core.Key'>' in '[6]: Mean/PerKey/CombinePerKey(MeanCombineFn)'. \n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "(Key(offsets={'lat': 0, 'lon': 0}, vars=None), <xarray.Dataset>\n",
+      "Dimensions:  (lat: 25, lon: 53)\n",
+      "Coordinates:\n",
+      "  * lat      (lat) float32 75.0 72.5 70.0 67.5 65.0 ... 25.0 22.5 20.0 17.5 15.0\n",
+      "  * lon      (lon) float32 200.0 202.5 205.0 207.5 ... 322.5 325.0 327.5 330.0\n",
+      "Data variables:\n",
+      "    air      (lat, lon) float64 260.4 260.2 259.9 259.5 ... 297.3 297.3 297.3)\n"
+     ]
+    }
+   ],
+   "source": [
+    "with beam.Pipeline() as p:\n",
+    "    p | xbeam.DatasetToChunks(ds, chunks={'time': 1000}) | xbeam.Mean('time') | beam.Map(print)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "65232829",
+   "metadata": {},
+   "source": [
+    "## Lower-level API"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "eabaf15a",
+   "metadata": {},
+   "source": [
+    "Xarray-Beam also includes lower-level transforations modelled off of [`beam.Mean`](https://beam.apache.org/documentation/transforms/python/aggregation/mean/) rather than {py:meth}`xarray.Dataset.mean`: they compute averages over sequences of `xarray.Dataset` objects or (`key`, `xarray.Dataset`) pairs, rather than calculating an average over an existing Xarray dimension or based on `xarray_beam.Key` objects, e.g.,"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "86a925af",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:apache_beam.options.pipeline_options:Discarding unparseable args: ['/Users/shoyer/miniconda3/envs/xarray-beam/lib/python3.9/site-packages/ipykernel_launcher.py', '-f', '/Users/shoyer/Library/Jupyter/runtime/kernel-b19331c2-e263-4737-bd64-012081154884.json']\n"
```

Comment:
Do you think it's worthwhile to remove this warning? It looks like Beam, in general, doesn't like how we're running the jupyter notebook (maybe, via conda?).

Reply:
The warnings don't show up in the rendered docs: https://xarray-beam--75.org.readthedocs.build/en/75/aggregation.html

I think there's a config setting somewhere that turned them off.

Reply:
I didn't know that! Good to know :)